	* glib/convert.c, glib/gmem.c: Do not crash when trying to convert
	a huge file, return NULL instead.

--- glib/gconvert.c.old	2007-01-26 17:16:07.171875000 +0100
+++ glib/gconvert.c	2007-01-26 17:17:35.812500000 +0100
@@ -610,7 +610,15 @@ g_convert_with_iconv (const gchar *str,
 		
 		outbuf_size *= 2;
 		dest = g_realloc (dest, outbuf_size);
-		
+
+		if (dest == NULL)
+		  {
+		    g_set_error (error, G_CONVERT_ERROR, G_CONVERT_ERROR_FAILED,
+			         _("Error during conversion: %s"),
+			         g_strerror (errno));
+		    return NULL;
+		  }
+
 		outp = dest + used;
 		outbytes_remaining = outbuf_size - used - 1; /* -1 for nul */
 	      }
--- glib/gmem.c.old	2007-01-26 17:10:51.218750000 +0100
+++ glib/gmem.c	2007-01-26 17:10:52.046875000 +0100
@@ -169,7 +169,7 @@ g_realloc (gpointer mem,
       if (mem)
 	return mem;
 
-      g_error ("%s: failed to allocate %lu bytes", G_STRLOC, n_bytes);
+      /* g_error ("%s: failed to allocate %lu bytes", G_STRLOC, n_bytes); */
     }
 
   if (mem)
